app/controllers/application_controller.rb

```diff
@@ -1,4 +1,25 @@
 # frozen_string_literal: true
 
 class ApplicationController < ActionController::Base
+  helper_method :current_user, :require_login
+
+  def index; end
+
+  def current_user
+    @current_user ||= User.find(session[:user_id]) if session[:user_id]
+  end
+
+  private
+
+  def require_login
+    if session[:user_id]
+      if current_user.nil?
+        flash[:error] = 'You must be logged in to access this section'
```

Comment:
This won't work, because `redirect_to` will discard content of the flash message.
If you want to redirect with flash, you need to do it as [docs says](https://api.rubyonrails.org/v7.0.3/classes/ActionController/Redirecting.html#method-i-redirect_to)